cloudferrylib/utils/files.py

```diff
@@ -89,4 +90,19 @@ def _unlink_dir(dirname):
         return "rm -rf {dir}".format(dir=dirname)
     else:
         raise RuntimeError('Wrong dirname %s, stopping' % dirname)
-    return
+
+
+class RemoteTempDir(object):
+    """Creates remote temp dir using `mktemp` and removes it on scope exit"""
+
+    def __init__(self, runner):
+        self.runner = runner
+
+    def __enter__(self):
+        create_temp_dir = 'mktemp -d'
```

Comment:
This one is different, because it creates dir, not only returns possible remote path. This is safer (no race conditions) and faster (single call needed for creation)